worker/deps/libwebrtc/libwebrtc/modules/congestion_controller/goog_cc/trendline_estimator.cc

```diff
@@ -127,56 +184,74 @@ TrendlineEstimator::TrendlineEstimator(
 
 TrendlineEstimator::~TrendlineEstimator() {}
 
+void TrendlineEstimator::UpdateTrendline(double recv_delta_ms,
+                                         double send_delta_ms,
+                                         int64_t send_time_ms,
+                                         int64_t arrival_time_ms,
+                                         size_t packet_size) {
+  const double delta_ms = recv_delta_ms - send_delta_ms;
+  ++num_of_deltas_;
+  num_of_deltas_ = std::min(num_of_deltas_, kDeltaCounterMax);
+  if (first_arrival_time_ms_ == -1)
+    first_arrival_time_ms_ = arrival_time_ms;
+
+  // Exponential backoff filter.
+  accumulated_delay_ += delta_ms;
+  // BWE_TEST_LOGGING_PLOT(1, "accumulated_delay_ms", arrival_time_ms,
+  //                      accumulated_delay_);
+  smoothed_delay_ = smoothing_coef_ * smoothed_delay_ +
```

Comment:
Worth to note here, It is a very strong / simplified / practical assumption here that the delay change follows linear trend, as near to the bw the changes spurious, and more like a second order linear regression could fit. However it could give worse results than this, but what I try to point out here is the error rate of the regression should be surface in order to use it in decisions at higher level. (the less it fits to a linear regression the more we can be sure there is a problem)

Reply:
Right now I want to build a dashboard which will allow to observe all the estimators values, and trendline estimator is my first candidate to observe because delay based estimator and loss v2 in some sense relies on it's values, and by only observing the log of how it's state changes I didn't liked it, but need visual representation to confirm it..

Reply:
I see, and makes sense. For validation purposes still make sense to measure the error rate, so how well the points fits to the.trend if the points are perfectly aligned then the error is 0, if they are very fuzzy around the assumed slope the error rate is high, which makes sense in validation too because it gives some information.

Reply:
So let me rephrase, in order to be sure that I understand:  You are telling that we should also calculate R squared here, and the farer it is from 0, the higher chances of we are overusing?

Reply:
No, not necessary. it is only makes sense together with the calculated trend. Higher error rate means higher fluctuation. if we don't have trend, then we probably have in this case high jitter and the playout should solve the problem. What might be a useful information from error rate for congestion control if the trend is not high is the presence of a concurrent flow. 

if we have trend and the error rate is low in this case you either have incipient congestion when the delay is being increased quasi in a linear fashion. I think majority of the cases the error rate will be high but what could be interesting to see if is there a correlation between the severity of the congestion and the error rate. We know for sure trend is high in case of congestion, but can we have an information how bad is the situation. Just a thought.